### PR TITLE
Add support for approver policy

### PR DESCRIPTION
**Pull request by SgtCoDFish**
If using cert-manager with the default approver disabled, it's a pain to install trust-manager since it requires manual approval of the CertificateRequest which is issued for the webhook.

This adds an option to create a CertificateRequestPolicy which approver-policy can use to auto-approve the CR.

This was tested with:

```bash
#!/usr/bin/env bash

set -exu -o pipefail

rm  ./bin/chart/trust-manager-v0.5.0.tgz && make chart
helm template trust-manager ./bin/chart/trust-manager-v0.5.0.tgz --namespace default --set app.trust.namespace=default --set approverPolicy.enabled=true > bin/templated.yaml

helm install \
	cert-manager jetstack/cert-manager \
	--namespace cert-manager \
	--create-namespace \
	--version v1.12.0 \
	--set installCRDs=true \
	--set extraArgs={--controllers='*\,-certificaterequests-approver'}

helm upgrade -i -n cert-manager cert-manager-approver-policy jetstack/cert-manager-approver-policy --wait

helm install trust-manager ./bin/chart/trust-manager-v0.5.0.tgz --namespace default --set app.trust.namespace=default --set approverPolicy.enabled=true

```

### CODE REVIEW

**Review by wallrj — APPROVED** [review]
Thanks @SgtCoDFish 

I didn't test it on my laptop, but the changes make sense and your testing notes in the PR description make sense.

Maybe the Helm values docs for `approverPolicy.enabled` could more clearly explain that it SHOULD (MUST?) be set to true if approver-policy is being used.
And / Or perhaps that can be added as a note to the trust-manager tutorial?

Do you intend to merge this **and** #157 ? Or will this get reverted if the long-lived Helm generated cert approach gets accepted? 

/lgtm
/hold so that you can merge this at your discretion.

File: deploy/charts/trust-manager/values.yaml
Comment (SgtCoDFish) [reply deploy/charts/trust-manager/values.yaml:19]: I'm happy to change it, I don't have a strong preference.

`app.webhook.tls.approverPolicy` could work. If that's the base, then #157 could become `app.webhook.tls.issueIsolatedCert` and we could also add `app.webhook.tls.disable`.

Does that work?

Comment (SgtCoDFish) [reply deploy/charts/trust-manager/values.yaml]: Following discussion in the cert-manager standup, I've now changed it to that!

**Review by inteon — COMMENTED** [review]
Thanks for the change.
One small question: I'm wondering how we could not create the cert and issuer in the chart  without eg. disabling tls.

**Comment by SgtCoDFish** [comment]
>  One small question: I'm wondering how we could not create the cert and issuer in the chart without eg. disabling tls.

#157 is one approach! I'm not sure how many more options it's worth adding though, I think the following list should be sufficient:

1. Generate in helm (#157 )
2. Generate via cert-manager (this is the only option today, and this PR just makes it work with approver-policy)
3. Disable TLS (future PR)

Possible to get an lgtm for this?

**Comment by inteon** [comment]
@SgtCoDFish I'm a bit worried about whether we can add those options and stay compatible with the current Helm structure, but I guess we will figure that out in the future.
/approve
/lgtm

**Pull request merged**
